getconf/base.py

```diff
@@ -169,6 +170,19 @@ def gettimedelta(self, key, default='0d', doc=''):
         else:
             return datetime.timedelta(**{unit: value})
 
+    def getpath(self, key, default=Path('.'), doc=''):
+        assert (
+            default is None or isinstance(default, (Path, str))
+        ), 'getpath("%s", %r) has an invalid default value type.' % (key, default)
```

Comment:
Note here: an `assert` might be optimized out when running with `python -O` or `python -OO`.

Reply:
Yes, all the `getXXX` method follow the same pattern. We might want to revisit it, but this would happen in an other PR.